### ISSUE TITLE
Ensure your project meets the CNCF IP Policy

### ISSUE THREAD

**Issue by csantanapr** [knative/community#952]
Ensure your project meets the CNCF IP Policy: https://github.com/cncf/foundation/blob/master/charter.md#11-ip-policy

Checklist of items

- [x] (a) Any project that is added to the CNCF must have ownership of its trademark and logo assets transferred to the Linux Foundation.
- [x] (b) Each project shall determine whether it will require use of an approved CNCF CLA. For projects that select to use a CLA, all code contributors will undertake the obligations set forth in the Apache contributor license agreement(s), altered only as necessary to identify CNCF as the recipient of the contributions, and which shall be approved by the Governance Board. See CNCF Contributor License Agreements available at https://github.com/cncf/cla. The process for managing contributions in accordance with this policy shall be subject to Governance Board approval.
- [x] (c) All new inbound code contributions to the CNCF shall be (i) accompanied by a Developer Certificate of Origin sign-off (https://developercertificate.org/) and (ii) made under the Apache License, Version 2.0 (available at https://www.apache.org/licenses/LICENSE-2.0), such license to be in addition to, and shall not supersede, obligations undertaken under the contribution license agreement(s) provided for in (b) above.
- [x] (d) All outbound code will be made available under the Apache License, Version 2.0.
- [x] (e) All projects evaluated for inclusion in the CNCF shall be completely licensed under an OSI-approved open source license. If the license for a project included in CNCF is not Apache License, Version 2.0, approval of the Governing Board shall be required.
- [x] (f) All documentation will be received and made available by the CNCF under the Creative Commons Attribution 4.0 International License.
- [x] (g) If an alternative inbound or outbound license is required for compliance with the license for a leveraged open source project or is otherwise required to achieve the CNCF’s mission, the Governing Board may approve the use of an alternative license for inbound or outbound contributions on an exception basis.

**Comment by csantanapr** [comment]
Related to [INCUBATING PROJECT ONBOARDING] Knative https://github.com/cncf/toc/issues/794

**Comment by csantanapr** [comment]
(a) Any project that is added to the CNCF must have ownership of its trademark and logo assets transferred to the Linux Foundation.

This depends on trademark being transferred tracking in https://github.com/knative/community/issues/948

**Comment by csantanapr** [comment]
(b) Each project shall determine whether it will require use of an approved CNCF CLA. For projects that select to use a CLA, all code contributors will undertake the obligations set forth in the Apache contributor license agreement(s), altered only as necessary to identify CNCF as the recipient of the contributions, and which shall be approved by the Governance Board. See CNCF Contributor License Agreements available at https://github.com/cncf/cla. The process for managing contributions in accordance with this policy shall be subject to Governance Board approval.
 (c) All new inbound code contributions to the CNCF shall be (i) accompanied by a Developer Certificate of Origin sign-off (https://developercertificate.org/) and (ii) made under the Apache License, Version 2.0 (available at https://www.apache.org/licenses/LICENSE-2.0), such license to be in addition to, and shall not supersede, obligations undertaken under the contribution license agreement(s) provided for in (b) above.

These depends on us selecting easyCLA or DCO first tracking here https://github.com/knative/community/issues/90

**Comment by csantanapr** [comment]
(d) All outbound code will be made available under the Apache License, Version 2.0.

This can be marked done all the repos license is Apache-2

**Comment by csantanapr** [comment]
All projects evaluated for inclusion in the CNCF shall be completely licensed under an OSI-approved open source license. If the license for a project included in CNCF is not Apache License, Version 2.0, approval of the Governing Board shall be required.

Is my understanding that the current set of licenses is clean as of today.

**Comment by csantanapr** [comment]
We don't have current issues today with this one
(g) If an alternative inbound or outbound license is required for compliance with the license for a leveraged open source project or is otherwise required to achieve the CNCF’s mission, the Governing Board may approve the use of an alternative license for inbound or outbound contributions on an exception basis.

**Comment by csantanapr** [comment]
(f) All documentation will be received and made available by the CNCF under the Creative Commons Attribution 4.0 International License.

We need to update the website repo to be dual license, any code should be under Apache-2 and any website rendered content under creative commons

**Comment by csantanapr** [comment]
I open an issue in docs repo to add the CC license for website https://github.com/knative/docs/issues/4878

**Comment by csantanapr** [comment]
(a) Any project that is added to the CNCF must have ownership of its trademark and logo assets transferred to the Linux Foundation.

Trademark has being officially transferred according to @thisisnotapril

**Comment by csantanapr** [comment]
What's left is implement the new CLA via easyCLA or DCO

**Comment by vaikas** [comment]
Is (b) done? We're going to use EasyCLA, so seems like b is done and we're working on c?
In particular b sez:
```
Each project shall determine whether it will require use of an approved CNCF CLA...
```

And we will require and it will be EasyCLA. So can we check the (b) done?

**Comment by csantanapr** [comment]
@knative/steering-committee met today and came up with the following

- b = Use a valid CLA can be DCO or easyCLA
- c = Any contributions need to follow the CLA from b and done under the Apache license

Action Items
1. Decide DCO vs. easyCLA, steering decided to go forward with easyCLA (done)
2. Replace Google CLA bot with easyCLA bot 

@lance will request easyCLA account for Knative and onboard all the current contributors into the account

**Comment by lance** [comment]
Isses opened in CNCF Jira/Help Desk
- https://cncfservicedesk.atlassian.net/servicedesk/customer/portal/1/CNCFSD-1235
- https://jira.linuxfoundation.org/plugins/servlet/theme/portal/4/SUPPORT-9913
Currently awaiting a response. I imagine CNCF folks are pretty busy with KubeCon EU happening next week.

**Comment by csantanapr** [comment]
@lance Can you log here the last status, I think is related to the bots getting approve by CNCF legal

**Comment by abrennan89** [comment]
@lance I can update the link included in our docs per https://github.com/knative/docs/issues/5047, however our docs page points to https://github.com/knative/community/blob/main/CONTRIBUTING.md#contributor-license-agreements, which should be updated by TOC/Steering/whoever owns this page. Please comment on the docs issue when this is done and I'll update the link, otherwise this will be inconsistent and confusing for users.

**Comment by lance** [comment]
> @lance I can update the link included in our docs per [knative/docs#5047](https://github.com/knative/docs/issues/5047), however our docs page points to https://github.com/knative/community/blob/main/CONTRIBUTING.md#contributor-license-agreements, which should be updated by TOC/Steering/whoever owns this page. Please comment on the docs issue when this is done and I'll update the link, otherwise this will be inconsistent and confusing for users.

👍 https://github.com/knative/community/pull/1102
